openmc/kinetics/solver.py

```diff
@@ -0,0 +1,831 @@
+        total = self.states['START'].absorption + self.states['START'].outscatter
+        f.create_dataset('total', data=total)
+
+        f.close()
+
+    def run_openmc(self, time_point, method='ADIABATIC'):
+
+        self.setup_openmc(time_point, method)
+
+        # Names of the statepoint and summary files
+        sp_old_name = '{}/statepoint.{}.h5'.format(self.directory, self.settings_file.batches)
+        sp_new_name = '{}/statepoint_{:.6f}_sec.{}.h5'\
+                      .format(self.directory, self.clock.times[time_point], self.settings_file.batches)
+        sum_old_name = '{}/summary.h5'.format(self.directory)
+        sum_new_name = '{}/summary_{:.6f}_sec.h5'.format(self.directory, self.clock.times[time_point])
+        old_source = '{}/source.{}.h5'.format(self.directory, self.settings_file.batches)
+        new_source = '{}/source.h5'.format(self.directory)
+
+        # Run OpenMC
+        if not self.use_pregenerated_sps:
+            openmc.run(threads=self.threads, mpi_args=self.mpi_args,
+                           cwd=self.directory)
+
+            # Rename the statepoint and summary files
+            copyfile(sp_old_name, sp_new_name)
+            copyfile(sum_old_name, sum_new_name)
+            move(old_source, new_source)
+
+        # Load the summary and statepoint files
+        summary_file = openmc.Summary(sum_new_name)
+        statepoint_file = openmc.StatePoint(sp_new_name, False)
+        statepoint_file.link_with_summary(summary_file)
+
+        # Load mgxs library
+        for mgxs in self.states[time_point].mgxs_lib.values():
+            mgxs.load_from_statepoint(statepoint_file)
+
+        self.states[time_point].load_mgxs()
+
+    def create_state(self, time_point):
+
+        if time_point in ['START', 'END', 'FORWARD_OUTER', 'PREVIOUS_OUTER']:
+            state = openmc.kinetics.OuterState(self.states)
+            state.chi_delayed_by_delayed_group = self.chi_delayed_by_delayed_group
+            state.chi_delayed_by_mesh = self.chi_delayed_by_mesh
+            if time_point in ['FORWARD_OUTER', 'PREVIOUS_OUTER']:
+                state.method = self.method
+        else:
+            state = openmc.kinetics.InnerState(self.states)
+            state.method = self.method
+
+        state.shape_mesh = self.shape_mesh
+        state.amplitude_mesh = self.amplitude_mesh
+        state.unity_mesh = self.unity_mesh
+        state.tally_mesh = self.tally_mesh
+        state.multi_group = self.multi_group
+        state.energy_groups = self.energy_groups
+        state.fine_groups = self.fine_groups
+        state.one_group = self.one_group
+        state.num_delayed_groups = self.num_delayed_groups
+        state.use_agd = self.use_agd
+        state.use_pcmfd = self.use_pcmfd
+        state.time_point = time_point
+        state.clock = self.clock
+        state.k_crit = self.k_crit
+        state.core_volume = self.core_volume
+        state.log_file = self.log_file
```

Comment:
Why not just have `state` keep a reference to the solver and get all these attributes from the solver when needed?